FLEDGE_implementation_overview.md

```diff
@@ -0,0 +1,64 @@
+
+This document starts out discussing singleton auctions, without any nested component auctions.  There is a section on component auctions at the end.
+
+
+## Phase 1 Load Interest Groups
+Immediately after an auction starts, all interest groups, k-anon data, and win history will be loaded from the SQLite database.  No worklets will be loaded until after this is complete, for all interestGroupBuyers participating in an auction.  Interest groups are loaded on a per-buyer basis, and remain grouped by buyer for the entire auction.  Interest groups that cannot participate in the auction (due to, e.g., no biddingLogicURL) are removed.  If there are no interest groups that can bid in the auction, and no additional bids, the auction completes without a winner, bypassing future stages.
+
+For each bidder origin, Chrome determines a tentative priority - using each interest group’s fixed priority, or priority vector, if present.  Chrome then sorts the interest groups based on priority, removing any with priority vectors and negative priorities.  Interest groups with matching priorities are grouped by joining origin (to improve performance in case ExecutionMode is groupByOrigin).  Groups with matching joining origins and priority are randomly shuffled.  For each bidder origin, Chrome checks if any interest group participating in the auction has “enableBiddingSignalsPrioritization” set.  If not, Chrome applies the auction’s per-buyer size limit to the bidder’s interest groups.  Either way, the interest groups are now in a priority order that will affect the order in which other resources will be fetched, and worklets will be run in.  None of this affects additional bids.
+
+
+## Phase 2 Bidding and Scoring Phase
+
+### Starting Worklet Executors
+Once all interest groups are loaded, Chrome starts loading the seller worklet, and all bidder worklets participating in an auction.  To load a worklet, Chrome first checks if there is already a matching worklet “executor” (Same worklet origin, same javascript URL, wasm URL, trusted signals URL, same frame - frame match is required by the DevTools hooks.  It may be for a different auction / component auction, however).  If so, the executor is reused.  Note that different executors do not currently share any network fetches or Javascript contexts, although they do share the same global HTTP cache, so it’s generally a good idea to try and use the same URLs for all interest groups, to maximize executor reuse.  Also, interest groups that share an executor are guaranteed to be run in the priority order they were sorted into (this may change if trusted signals fetches are split up, though).  Interest groups that don’t share an executor are run on that basis of whichever has everything it needs to run first, runs first.
```

Comment:
Top-level user-visible frame?

Reply:
Nope - actually means the frame runAdAuction() was called in, unfortunately.  Clarified.